### PR TITLE
Provide right git url in purs publish error message

### PR DESCRIPTION
**Pull request by coot**
cc @hdgarrood

### CODE REVIEW

**Comment by hdgarrood** [comment]
In many cases the result of `git config remote.origin.url` is going to be an SSH url, and I don't think we should suggest that, because then I think people can only clone it if they have a public key which GitHub (or whatever host) recognises. Also, many networks block SSH and the Git protocol, but HTTPS works pretty much everywhere. So I think we should only ever suggest HTTPS urls. We could automatically rewrite GitHub SSH urls to GitHub HTTPS urls to avoid this, though.

Also, the main purpose of that hint is so that people don't have to go and look up how to put this information into bower.json, so I think that hint should be displayed every time, even if we do have to use a default in some cases.

How about this:

- if it's an HTTPS url, suggest it as-is
- if it's an SSH or Git url and hosted on GitHub, rewrite it to an HTTPS url and suggest that
- if it's any other kind of url, ignore it and display the hint with a default url: `https://github.com/myuser/myrepo.git` - it should then be more obvious than it is currently that you should change it.

**Comment by coot** [comment]
I'd just capitalize: `https://github.com/USER/REPO.git` so it's easy to spot.

**Comment by hdgarrood** [comment]
Good idea :+1:

**Comment by coot** [comment]
@hdgarrood it's ready for review.

File: src/Language/PureScript/Publish/ErrorsWarnings.hs [hunk 1]
```diff
@@ -234,19 +234,18 @@ spdxExamples =
...
 displayRepositoryError :: RepositoryFieldError -> Box
 displayRepositoryError err = case err of
-  RepositoryFieldMissing ->
+  RepositoryFieldMissing giturl ->
     vcat
       [ para (concat
          [ "The 'repository' field is not present in your package manifest file. "
          , "Without this information, Pursuit would not be able to generate "
-         , "source links in your package's documentation. Please add one - like "
-         , "this, for example:"
```
Comment (hdgarrood) [inline src/Language/PureScript/Publish/ErrorsWarnings.hs]: I think I prefer the original text here, actually. We don't want to be too prescriptive: people might have a reason for having `remote.origin.url` differ from what's inside bower.json.

File: src/Language/PureScript/Publish.hs [hunk 2]
```diff
@@ -202,15 +202,20 @@ getTagTime tag = do
...
   where
-  tryExtract pkgMeta =
-    case bowerRepository pkgMeta of
-      Nothing -> Left RepositoryFieldMissing
-      Just Repository{..} -> do
-        unless (repositoryType == "git")
-          (Left (BadRepositoryType repositoryType))
-        maybe (Left NotOnGithub) Right (extractGithub repositoryUrl)
+    format :: (D.GithubUser, D.GithubRepo) -> Text
+    format (user, repo) = "https://github.com/" <> (D.runGithubUser user) <> "/" <> (D.runGithubRepo repo) <> ".git"
```
Comment (hdgarrood) [inline src/Language/PureScript/Publish.hs]: Sorry to nitpick but could you remove the redundant parens here, please?

**Pull request merged**